### PR TITLE
feat(useExhaustiveDependencies): add option to disable errors for unecessary dependencies

### PR DESCRIPTION
**Pull request by simon-paris**
Closes https://github.com/biomejs/biome/issues/630

## Summary

This adds an option to suppress errors for dependencies that are in the dependencies array but not used in the hook function.

The option schema is below. It could also have the new field inside the hooks array but I think you'd generally want to use this for every hook type.

```
// this config
"useExhaustiveDependencies": {
  "reportUnnecessaryDependencies": false,
  "hooks": [...] // unchanged
}

// allows this
function ReportUnecessaryDependencies() {
    const [a] = useState("hello")
    const [b] = useState("world");
    useEffect(() => {
        console.log(a); // <- prints when b changes, but does not use b
    }, [a, b]);
    return a;
}
```

Additionally, the options struct for this rule was originally designed to be reused, but at some point it was removed from useHookAtTopLevel [here](https://github.com/biomejs/biome/issues/630), now it's not actually reused anywhere. Since this PR adds an option field specific to this rule, I've renamed the options struct to indicate that it's non-reusable.

## Test Plan

A unit test is included

### CODE REVIEW

File: crates/biome_js_analyze/src/lint/correctness/use_exhaustive_dependencies.rs [hunk 1]
```diff
@@ -289,15 +289,29 @@ impl Default for ReactExtensiveDependenciesOptions {
...
 
 /// Options for the rule `useExhaustiveDependencies`
-#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[derive(Clone, Debug, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
 #[cfg_attr(feature = "schemars", derive(JsonSchema))]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct HooksOptions {
+pub struct UseExhaustiveDependenciesOptions {
+    /// Whether to report an error when a dependency is listed in the dependencies array but isn't used. Defaults to true.
+    #[serde(default)]
```
Comment (arendjr) [inline crates/biome_js_analyze/src/lint/correctness/use_exhaustive_dependencies.rs:297]:
```suggestion
    #[serde(default = "report_unnecessary_dependencies_default")]
```

And then later:

```rs
fn report_unnecessary_dependencies_default() -> bool {
    true
}
```

The `Default` implementation on `UseExhaustiveDependenciesOptions` is only used when calling `UseExhaustiveDependenciesOptions::default()` (for instance, when _no_ options have been specified at all), so it is not sufficient to make Serde respect the defaults of individual fields.

I would then update the `Default` impl as well:

```rs
impl Default for UseExhaustiveDependenciesOptions {
    fn default() -> Self {
        Self {
            report_unnecessary_dependencies: report_unnecessary_dependencies_default(),
            hooks: vec![],
        }
    }
}
```

**Review by arendjr — COMMENTED** [review]
Nice work! Just need to fix the defaults a bit.

**Comment by simon-paris** [comment]
@arendjr Thanks, all fixed

**Review by arendjr — APPROVED** [review]

**Comment by arendjr** [comment]
Just realized we forgot a CHANGELOG entry. Feel free to add it in the next PR or do a separate one.

**Pull request merged**